app/frontend/src/cljs/teet/ui/query.cljs

```diff
@@ -46,38 +49,50 @@
         previous-args (atom args)
         state-atom (when-not state-path
                      (r/atom nil))
+        filter-atom (r/atom {})
+        debounce (functions/debounce (fn []
+                                       (e! (common-controller/->Refresh)))
+                                     250)
+        change-filter #(swap! filter-atom merge %)
+        reset-filter #(reset! filter-atom {})
         poll-id (when poll-seconds
                   (js/setInterval #(e! (->Query query args state-path state-atom))
                                   (* 1000 poll-seconds)))]
+    (add-watch filter-atom :requery-filters
+               (fn [_ _ _ _]
+                 (debounce)))
```

Comment:
debounce isn't very descriptive, should the fn be called `refresh!` or something like that